pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -303,18 +303,31 @@ func (r *ReconcileMasterUserRecord) deleteUserAccount(logger logr.Logger, target
 	// Get the User associated with the UserAccount
 	userAcc := &toolchainv1alpha1.UserAccount{}
 	namespacedName := types.NamespacedName{Namespace: memberCluster.OperatorNamespace, Name: name}
-	err = memberCluster.Client.Get(context.TODO(), namespacedName, userAcc)
-	if err != nil {
+	if err = memberCluster.Client.Get(context.TODO(), namespacedName, userAcc); err != nil {
 		if errors.IsNotFound(err) {
 			return nil
 		}
 		return err
 	}
+
+	if coputil.IsBeingDeleted(userAcc) {
+		return fmt.Errorf("UserAccount deletion in progress")
```

Comment:
This state is not an error, I would just return here with postponed reconciliation. WDYT?

Reply:
Yeah, I thought about this too but the main reason I wanted to return it as an error is because if there's a problem deleting the UserAccount and it's stuck deleting then returning the error would make use of the controller's backoff mechanism so that we're not reconciling every x seconds when there's a persistent failure.

Reply:
ok, how about doing this - let's check the deletion timestamp and if it's older than 1 minute, then start returning an error, until then, just reconcile every 10 seconds.

Reply:
That's a good suggestion! 🙂

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/352/commits/e9a3765ce783f8eb8561e95d79636723ef9fc666